applications/serial_lte_modem/src/slm_util.c

```diff
@@ -143,3 +145,31 @@ bool check_for_ipv4(const char *address, uint8_t length)
 
 	return true;
 }
+
+/**brief use AT command to get IPv4 address
+ */
+bool util_get_ipv4_addr(char *address)
+{
+	int err;
+	char at_rsp[128];
+	char *tmp1, *tmp2;
+
+	err = at_cmd_write("AT+CGPADDR", at_rsp, sizeof(at_rsp), NULL);
+	if (err) {
+		return false;
+	}
+	/* parse +CGPADDR: 0,"10.145.192.136" */
+	tmp1 = strstr(at_rsp, "\"");
+	if (tmp1 == NULL) {
+		return false;
+	}
+	tmp1++;
+	tmp2 = strstr(tmp1, "\"");
+	if (tmp2 == NULL) {
+		return false;
+	}
+
+	memset(address, 0x00, NET_IPV4_ADDR_LEN);
+	strncpy(address, (const char *)tmp1, (size_t)(tmp2 - tmp1));
```

Comment:
I think `memcpy` would be more suitable here, IP address pointed by tmp1 isn't really a C-string (isn't NULL terminated). And you know the exact size of the data to copy:
```
memcpy(address, tmp1, tmp2 - tmp1);
address[tmp2 - tmp1] = '/0'; /* NULL termination or the IP string. */
```

Reply:
Great suggestion, taken with gratitude.